Yoast/javascript
# Make new internal linking separate functions

## Summary

* [non-user-facing] Makes the new internal linking functionality available through separate researches, stringProcessing functions and a class `ProminentWord`, in order to facilitate the introduction of the feature-flag functionality that will allow to switch between old and new ways.
* [non-user-facing] Makes naming of internal-linking-related files and functions more consistent within the `yoastseo` package.
* [non-user-facing] Adjusts the `content-analysis` app to display prominent words for old analysis, new analysis for insights and new analysis for internal linking.

## Relevant technical choices:

* I chose to split old and new processing streams completely. I.e., I put the old `relevantWords` research back, all functions that it calls and all relevant specs. The new functionality is made available through two new researches, `getProminentWordsForInsights` (added in previous PRs) and `getProminentWordsForInternalLinking`, which both call a stringProcessing function `determineProminentWords`, which creates objects of type `ProminentWord` (as opposed to the old researches that return `WordCombination`s)
* For consistency of naming, the newly introduced stringProcessing functions were renamed as follows (Might be handy when editing the components and the `content-analysis` app):
`getRelevantCombinations` -> `filterProminentWords`
`sortCombinations` - > `sortProminentWords`
`collapseRelevantWordsOnStem` -> `collapseProminentWordsOnStem`
`getRelevantWords` -> `getProminentWords`
`getRelevantWrodsFromPaperAttributes` - > `getProminentWrodsFromPaperAttributes`

## Test instructions

This PR can be tested by following these steps:

* See https://github.com/Yoast/wordpress-seo-premium/pull/2343 for testing instructions.
* As a sanity check: compare this branch to `develop`. All changes should fall under `adding some new functionality`, not `changing old functionality`.
* `content-analysis` dev tool: run as usual and check that the old analysis returns multi-word combinations with additional metrics, while the new analysis returns single words with stems and number of occurrences only.

## UI changes

## Quality assurance


Fixes #204
